### PR TITLE
dqlite: Use `go-dqlite` v3 and build libdqlite from master branch

### PR DESCRIPTION
**Pull request by roosterfish**

### CODE REVIEW

**Comment by roosterfish** [comment]
@kadinsayani is there a specific reason for running the unit and system tests as root? Is this required by TICS?

**Comment by kadinsayani** [comment]
> @kadinsayani is there a specific reason for running the unit and system tests as root? Is this required by TICS? 

It's not required by TICS. The only requirement is the presence of GOCOVERDIR.

**Review by kadinsayani — APPROVED** [review]
LGTM! Thanks!

**Review by MggMuggins — APPROVED** [review]
Looks great! I'd love to see this get merged so that we can implement the equivalent of https://github.com/canonical/lxd/pull/14378

File: go.mod [hunk 1]
```diff
@@ -20,6 +20,7 @@ require (
 require (
 	github.com/Rican7/retry v0.3.1 // indirect
 	github.com/armon/go-proxyproto v0.1.0 // indirect
+	github.com/canonical/go-dqlite v1.22.0 // indirect
```
Comment (tomponline) [inline go.mod:23]: @roosterfish what is left that is depending on go-dqlite v1?

Comment (roosterfish) [reply go.mod:23]: I am pretty sure that comes from the old version of LXD before merging the dqlite v2 PR.

Comment (roosterfish) [reply go.mod:23]: Should be fixed after we track `@stable-5.21`.

Comment (tomponline) [reply go.mod:23]: But you're not changing this branch to track `@stable-5.21` so should already be gone by updating the lxd dep?

Comment (roosterfish) [reply go.mod:23]: The LXD dep used here was still an older one before LXD was switched to dqlite v2.

**Pull request merged**